build(docs-infra): update @angular/{cli,core,material} to latest versions and TS to 3.8

gkalpak opened the pull request:
☝️ 

Jira issue: [FW-1970](https://angular-team.atlassian.net/browse/FW-1970)

##
Blocking issues:

- [ ] Confirm that the payload size increases are acceptable.
- [x] ~~This seems to be very prone to [FW-2008][1].
    We should address FW-2008 first, before merging this.~~

[1]: https://angular-team.atlassian.net/browse/FW-2008

[FW-2008]: https://angular-team.atlassian.net/browse/FW-2008

mary-poppins commented:
You can preview 6d18a79 at https://pr36145-6d18a79.ngbuilds.io/.

mary-poppins commented:
You can preview 91e4a21 at https://pr36145-91e4a21.ngbuilds.io/.

mary-poppins commented:
You can preview 7a3fc1f at https://pr36145-7a3fc1f.ngbuilds.io/.

filipesilva commented on aio/scripts/_payload-limits.json:
```diff
@@ -3,7 +3,7 @@
     "master": {
       "uncompressed": {
         "runtime-es2015": 2987,
-        "main-es2015": 450612,
```
So the size begins at 450676. Updating CLI bumps it to 450677, then updating FW bumps it to 451655, then updating material bumps it to 451935.

gkalpak replied:
Exactly! Thx for summarizing the changes :+1:

mary-poppins commented:
You can preview 3d6d785 at https://pr36145-3d6d785.ngbuilds.io/.

mary-poppins commented:
You can preview 4030e1d at https://pr36145-4030e1d.ngbuilds.io/.

mary-poppins commented:
You can preview 0960d22 at https://pr36145-0960d22.ngbuilds.io/.

gkalpak committed:
build(docs-infra): update angular.io payload sizes for reference

This commit updates all payload sizes for angular.io to make it easier
to compare payload size changes as a result of upgrading Angular
packages and other dependencies in subsequent commits.

gkalpak committed:
build(docs-infra): update @angular/cli to 10.0.0-next.3

Update the Angular CLI and Angular framework packages to latest `@next`
versions. Also, update the app to look more closely to how a newly
generated app with the latest CLI would look like.

gkalpak committed:
build(docs-infra): update @angular/material to 9.2.2

This commit updates the Angular Material packages (`@angular/cdk` and
`@angular/material`) to latest versions.

gkalpak committed:
ci(docs-infra): remove redundant standalone ngcc run

Previously, in the `test_aio` CI job, we ran ngcc before building the
app with `yarn build`. This was supposed to have the benefit of taking
advantage of the parallel capabilities of standalone ngcc (vs implicitly
running it via `ng build`).

It turns out that the work done by the standalone ngcc was thrown away
before the `ng build`, resulting in `ng build` having to run ngcc all
over again. This happened because the `yarn build` script (run after the
standalone ngcc step) also runs `yarn install`, which essentially cleans
up `node_modules/`, thus discarding all the work already done by ngcc.

Here is an [example CI job][1], where this can be seen in action:
One can see the "Compiling <some-package> : es2015 as esm2015" logs in
the `yarn --cwd aio ngcc --properties es2015` step (as the standalone
ngcc processes the various entry-points) and then see the same logs in
the `yarn --cwd aio build --progress=false` step (as ngcc has to process
the entry-points all over again).

This commit removes the redundant standalone ngcc run and lets the CLI
handle ngcc via `ng build`. It is possible to instrument the build
process in a way that we can run the standalone ngcc after
`yarn install` and thus take advantage of the performance gains in
parallel mode, but the latest version of the CLI can already run ngcc in
parallel mode as a pre-build step, so this is unnecessary.

[1]: https://circleci.com/gh/angular/angular/658691

gkalpak committed:
build(docs-infra): update TypeScript to 3.8

Update TypeScript for angular.io to the latest stable version: 3.8.3

Jira issue: [FW-1970](https://angular-team.atlassian.net/browse/FW-1970)

mary-poppins commented:
You can preview ede9afb at https://pr36145-ede9afb.ngbuilds.io/.

gkalpak committed:
build(docs-infra): upgrade docs examples to latest Angular CLI and framework

This commit also updates the projects to more closely match what a newly
generated app would look like with the exception of `tslint.json` files,
which would create too many linting failures. These will be updated in a
follow-up PR.

gkalpak committed:
fixup! build(docs-infra): update @angular/cli to 10.0.0-next.3

mary-poppins commented:
You can preview 7b56d1d at https://pr36145-7b56d1d.ngbuilds.io/.

gkalpak committed:
fixup! build(docs-infra): update @angular/material to 9.2.2

mary-poppins commented:
You can preview 9e05808 at https://pr36145-9e05808.ngbuilds.io/.

mary-poppins commented:
You can preview 66d9c41 at https://pr36145-66d9c41.ngbuilds.io/.

kara commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,26 +3,26 @@
         "runtime-es2015": 2987,
-        "main-es2015": 450612,
-        "polyfills-es2015": 52195
+        "main-es2015": 454032,
```
Oof, looks like this got worse since the last time we updated this PR...

kara reviewed (approved):
Not happy about the size regression, but given that this PR doesn't cause it and this is blocking other things, LGTM. 

I'll create a Jira to investigate the regression separately.

petebacondarwin reviewed (approved):
But I don't understand why the examples are left at 9.1.x while the rest of AIO is updated to 10.0.0-next.x

gkalpak commented:
> But I don't understand why the examples are left at 9.1.x while the rest of AIO is updated to 10.0.0-next.x

Examples are what people will use to play with (either on StackBlitz or locally), so we generally stick to stable versions for examples. The angular.io app itself can be on any version we are happy with (and currently we need 10-next, because it is the only one that contains #36626 until the next 9.x release).
